app/javascript/packs/search.js

```diff
@@ -1,14 +1,62 @@
 function textEntered(field) {
-  if (!field) {
-    field = document.getElementById("search");
-  }
-  localStorage.setItem('query', field.value ?? '');
-  const btn = document.getElementById("cancel");
-  if (field.value === "") {
+    clearTimeout(this.timeout)
+    let search = true;
+    if (!field) {
+        field = document.getElementById("search");
+        search = false;
+    }
+    localStorage.setItem('query', field.value ?? '');
+    const btn = document.getElementById("cancel");
+    if (field.value === "") {
+        btn.style.visibility = "hidden";
+    } else {
+        btn.style.visibility = "visible";
+    }
```

Comment:
This method feels too hard to understand. One idea would be to extract this part into an `updateClearButtonStyle` function

Reply:
done